tests/Doctrine/Tests/Common/Cache/Psr6/CacheAdapterTest.php

```diff
@@ -127,4 +128,40 @@ public function testItemsAreFlushedToTheUnderlyingCacheOnce(): void
         $adapter->commit();
         $adapter->commit();
     }
+
+    /**
+     * @dataProvider doctrineProviderFactoryProvider
+     * @psalm-param Closure(CacheItemPoolInterface): (DoctrineProvider|SymfonyDoctrineProvider) $doctrineProviderFactory
+     */
+    public function testNamespacingFeatureIsPreservedWithDoctrineProvider(Closure $doctrineProviderFactory): void
+    {
+        $wrapped = new ArrayAdapter();
+
+        $cacheApp1 = $doctrineProviderFactory($wrapped);
+        $cacheApp1->setNamespace('app 1');
+
+        $cacheApp2 = $doctrineProviderFactory($wrapped);
+        $cacheApp2->setNamespace('app 2');
+
+        $psrCacheApp1 = CacheAdapter::wrap($cacheApp1);
+        $psrCacheApp2 = CacheAdapter::wrap($cacheApp2);
+
+        $item = $psrCacheApp1->getItem('some key')->set('some value');
+        $psrCacheApp1->save($item);
+        self::assertFalse($psrCacheApp2->getItem('some key')->isHit());
+    }
+
+    /**
+     * @psalm-return iterable<string, array{Closure(CacheItemPoolInterface): (Cache|SymfonyDoctrineProvider)}>
+     */
+    public function doctrineProviderFactoryProvider(): iterable
+    {
+        yield 'Doctrine' => [Closure::fromCallable([DoctrineProvider::class, 'wrap'])];
```

Comment:
The only flaw I see here is that we don't know for sure that this closure is going to return a `DoctrineProvider`… and there does not seem to be a way to get one for sure: https://github.com/doctrine/cache/blob/cebb0d13f607775ff4cca587995558dce87a026c/lib/Doctrine/Common/Cache/Psr6/DoctrineProvider.php#L31-L47

Reply:
I have ditched the data provider I was using because the SymfonyDoctrineProvider no longer exists when using PHP 8, but the issue is still here: `DoctrineProvider::wrap` may not return a `DoctrineProvider`

Reply:
I just realised we don't have static analysis setup on this project and probably never will, so this shouldn't be an issue.

Reply:
and that's fine. `DoctrineProvider::wrap` is *not* expected to return a DoctrineProvider but a Cache.